engine/consensus/matching/engine.go

```diff
@@ -620,19 +633,35 @@ RES_LOOP:
 			return nil, fmt.Errorf("could not determine chunk assignment: %w", err)
 		}
 
+		matched := false
```

Comment:
since the default is `false`, if `len(incorporatedResult.Result.Chunks) == 0`, then matched should be `true`, but actually is `false`

better to handle this special case.

Reply:
sounds good 👍  Thanks for the detailed reasoning.